src/test/java/com/google/devtools/build/lib/rules/cpp/Cpp20ModulesConfiguredTargetTest.java

```diff
@@ -0,0 +1,157 @@
+package com.google.devtools.build.lib.rules.cpp;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.devtools.build.lib.analysis.util.AnalysisMock;
+import com.google.devtools.build.lib.analysis.util.BuildViewTestCase;
+import com.google.devtools.build.lib.cmdline.LabelSyntaxException;
+import com.google.devtools.build.lib.packages.util.Crosstool;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.IOException;
+
+@RunWith(JUnit4.class)
+public class Cpp20ModulesConfiguredTargetTest extends BuildViewTestCase {
+  @Before
+  public void setupBasicRulesWithModules() throws IOException {
+    scratch.file(
+        "foo/BUILD",
+        """
+        cc_library(
+            name = 'lib',
+            module_interfaces = ["foo.cppm"],
+        )
+        cc_binary(
+            name = 'bin',
+            module_interfaces = ["foo.cppm"],
+        )
+        cc_test(
+            name = 'test',
+            module_interfaces = ["foo.cppm"],
+        )
+        """);
+  }
+  @Test
+  public void testCpp20ModulesConfigurationNoFlags() throws LabelSyntaxException {
+    {
```

Comment:
I'd propose to split this one into 3 separate tests and similarly the next one. Otherwise, I'd be concerned if `assertContainsEvent` works on the current target or on the previously tested one.